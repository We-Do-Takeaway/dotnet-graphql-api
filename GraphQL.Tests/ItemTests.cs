using System.Threading.Tasks;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Xunit;

namespace WeDoTakeawayAPI.GraphQL.Tests
{
    public class ItemTests : BaseTests
    {
        [Fact]
        public async Task Get_Item_By_Id()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query ItemById($id: ID!) {
                          itemById(id: $id) {
                            id
                            name
                          }
                        }")
                    .SetVariableValue(name: "id", value: "600dca30-c6e2-4035-ad15-783c122d6ea6")
                    .Create());
        
            result.MatchSnapshot();
        }

        [Fact]
        public async Task Get_Item_Sections()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                .New()
                .SetQuery(@"
                    query ItemById($id: ID!) {
                      itemById(id:$id) {
                        id
                        name
                        sections {
                          id
                          name
                        }
                      }
                    }")
                .SetVariableValue(name: "id", value: "600dca30-c6e2-4035-ad15-783c122d6ea6")
                .Create());
        
            result.MatchSnapshot();
        }
        
        [Fact]
        public async Task Get_Item_Ingredients()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                    query ItemById($id: ID!) {
                      itemById(id:$id) {
                        id
                        name
                        ingredients {
                            id
                            name
                            quantity
                        }
                      }
                    }")
                    .SetVariableValue(name: "id", value: "600dca30-c6e2-4035-ad15-783c122d6ea6")
                    .Create());
        
            result.MatchSnapshot();
        }
    }
}