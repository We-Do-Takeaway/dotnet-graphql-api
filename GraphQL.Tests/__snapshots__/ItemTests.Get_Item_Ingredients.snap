{
  "Label": null,
  "Path": null,
  "Data": {
    "itemById": {
      "id": "600dca30-c6e2-4035-ad15-783c122d6ea6",
      "name": "Bowl of cherries",
      "ingredients": [
        {
          "id": "600dca30-c6e2-4035-ad15-783c122d6ea8",
          "name": "Cherry",
          "quantity": 2
        }
      ]
    }
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
